.phpcs.config.xml

```diff
@@ -2,8 +2,8 @@
 <!-- This file configures everything except the list of rules. -->
 <!-- The separation is so .github/files/phpcompatibility-dev-phpcs.xml can use the same config with a different rule set. -->
 <ruleset>
-	<config name="minimum_supported_wp_version" value="6.2" />
-	<config name="testVersion" value="5.6-"/>
+	<config name="minimum_supported_wp_version" value="6.3" />
+	<config name="testVersion" value="7.0-"/>
```

Comment:
:tada: 

I'm guessing a separate PR will be for dropping obsolete code paths, but if we're going to do this here then let's go ahead and drop PHP 5 from other places in CI too:

* https://github.com/Automattic/jetpack/blob/133f8dbf11f8506a7bf1f9274b44e98fe738357c/.github/files/generate-ci-matrix.php#L71-L78
* https://github.com/Automattic/jetpack/blob/133f8dbf11f8506a7bf1f9274b44e98fe738357c/.github/versions.sh#L8
* https://github.com/Automattic/jetpack/blob/133f8dbf11f8506a7bf1f9274b44e98fe738357c/.github/workflows/linting.yml#L174

Bunch of doc references to 5.6 too:

* https://github.com/Automattic/jetpack/blob/133f8dbf11f8506a7bf1f9274b44e98fe738357c/docs/coding-guidelines.md?plain=1#L7
* https://github.com/Automattic/jetpack/blob/133f8dbf11f8506a7bf1f9274b44e98fe738357c/docs/development-environment.md?plain=1#L378-L384
* https://github.com/Automattic/jetpack/blob/133f8dbf11f8506a7bf1f9274b44e98fe738357c/docs/monorepo.md?plain=1#L45
* https://github.com/Automattic/jetpack/blob/133f8dbf11f8506a7bf1f9274b44e98fe738357c/docs/monorepo.md?plain=1#L213
* https://github.com/Automattic/jetpack/blob/133f8dbf11f8506a7bf1f9274b44e98fe738357c/docs/monorepo.md?plain=1#L221
* https://github.com/Automattic/jetpack/blob/133f8dbf11f8506a7bf1f9274b44e98fe738357c/docs/testing/regression-checklist/regression-checklist.md?plain=1#L24

Reply:
> I'm guessing a separate PR will be for dropping obsolete code paths,

Yup, doing this in #34126